src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -758,12 +748,11 @@ public void close() {
 
 		private void onOnline() {
 			synchronized (this.thisLock) {
-				if (this.isOnline) {
+				if (this.isOnline()) {
```

Comment:
>isOnline [](start = 13, length = 8)

Is it ok to reacquire the same synchonized lock in Java? (This method synchronizes on thisLock and calls isOnline which also then synchronizes on thisLock).

Reply:
If you are worried about it creating a deadlock, I have tested it nd it won't happen.

---
In reply to: [260939178](https://github.com/Azure/azure-relay-java/pull/25#discussion_r260939178) [](ancestors = 260939178)

Reply:
good that it won't deadlock (the lock is re-entrant). I general I try to avoid asking for a lock while already owning that lock.

---
In reply to: [260945799](https://github.com/Azure/azure-relay-java/pull/25#discussion_r260945799) [](ancestors = 260945799,260939178)